src/util/tracking.ts

```diff
@@ -152,19 +152,25 @@ export function trackError(
     [key: string]: any
   }
 ): void {
-  let err: Error | string
-  if (error instanceof Error || typeof error === 'string') {
+  let err: Error
+  if (error instanceof Error) {
     err = error
+  } else if (typeof error === 'string') {
+    err = new Error(error)
   } else {
-    // At least send an error which should give us the callstack
-    err = 'Unknown error occurred'
+    // At least send an error which should give us the call-stack
+    err = new Error(`Unknown error occurred: ${String(error)}`)
   }
 
-  if (tag == null) {
-    captureException(err)
-  } else {
-    captureException(err, { event_id: tag, data: metadata })
+  let hint: { event_id?: string; data?: { [key: string]: any } } | undefined
+  if (tag != null) {
+    hint = { event_id: tag }
+  }
+  if (metadata != null) {
+    hint = { data: metadata }
```

Comment:
This doesn't let us have *both* a tag and a metadata at the same time. Maybe do something like this:

```ts
let hint: { event_id?: string; data?: { [key: string]: any } } = {}
if (tag != null) hint.event_id = tag
if (metadata != null) hint.data = metadata
captureException(err, tag != null || metadata != null ? hint : undefined)
```

Reply:
Oh, I see that the next PR completely re-does all this stuff, so nevermind.

Reply:
I think you mean to retract this feedback then. I'll resolve.